graphql_compiler/macros/macro_expansion.py

```diff
@@ -146,48 +154,43 @@ def _merge_selection_into_target(target_ast, selection_ast, subclass_sets=None):
     continuation_ast = selection_ast
     if coercion is not None:
         coercion_class = coercion.type_condition.name.value
+        continuation_ast = coercion
         if isinstance(target_ast, InlineFragment):
-            target_class = target_ast.type_condition.name.value
-            continuation_ast = coercion
             target_ast.type_condition = coercion.type_condition
-
-            # coercion_class is required to subclass target_class so we can merge the type coercions
-            if coercion_class != target_class:
-                if subclass_sets is None:
-                    # TODO(bojanserafimov): Write test for this failure
-                    raise GraphQLCompilationError(
-                        u'Cannot prove type coercion at macro target is valid. Please provide a '
-                        u'hint that {} subclasses {} using the subclass_sets argument.'
-                        .format(coercion_class, target_class))
-                else:
-                    if (target_class not in subclass_sets or
-                            coercion_class not in subclass_sets[target_class]):
-                        raise GraphQLCompilationError(u'Invalid type coercion at macro target. {} '
-                                                      u'is expected to subclass {}.'
-                                                      .format(coercion_class, target_class))
         else:
-            # TODO(bojanserafimov): Compute the type at the target and use it to validate.
-            raise NotImplementedError(u'Cannot coerce from macro that does not end '
-                                      u'in a coercion.')
+            target_ast.selection_set = SelectionSet([InlineFragment(
+                coercion.type_condition, target_ast.selection_set, directives=[])])
```

Comment:
This is a bit difficult to read. Maybe construct the `InlineFragment` separately and then pass it to the `SelectionSet([...])`?